### PR TITLE
MultiplAnalogSensorsClient: add --externalConnection option

### PR DESCRIPTION
**Pull request by Nicogene**
TESTED

Please @traversaro check if I am missing something

### CODE REVIEW

File: src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.h [hunk 1]
```diff
@@ -46,11 +46,12 @@ class SensorStreamingDataInputPort :
...
+* |:------------------:|:--------------:|:-------:|:--------------:|:-------------:|:------------:|:--------------------------------------------------------------------------------------:|:-----:|
+* | remote             |       -        | string  | -              |   -           | Yes          | Prefix of the ports to which to connect, opened by MultipleAnalogSensorsServer device. |       |
+* | local              |       -        | string  | -              |   -           | Yes          | Port prefix of the ports opened by this device.                                        |       |
+* | timeout            |       -        | double  | seconds        | 0.01          | No           | Timeout after which the device reports an error if no measurement was received.        |       |
+* | externalConnection |       -        | bool    | -              | false         | No           | If set to true, the connection to the rpc port of the MAS server is skipped and it is possible to connect to the data source externally after being opened | Use case: e.g yarpdataplayer source |
```
Comment (traversaro) [inline src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.h]: Probably it is worth to mention that in this case getName methods will fail

**Comment by traversaro** [comment]
cc @prashanthr05 this may be useful for the use case "run estimator over offline data".

**Comment by Nicogene** [comment]
I successfully tested this patch with the `iCubGui`patch(https://github.com/robotology/icub-main/pull/596)

Ready to be merged.

Waiting green light @traversaro

File: src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.cpp [hunk 2]
```diff
@@ -142,7 +145,7 @@ bool MultipleAnalogSensorsClient::close()
 
 size_t MultipleAnalogSensorsClient::genericGetNrOfSensors(const std::vector<SensorMetadata>& metadataVector) const
 {
-    return metadataVector.size();
+    return m_externalConnection ? 0 : metadataVector.size();
```
Comment (traversaro) [inline src/devices/multipleanalogsensorsclient/MultipleAnalogSensorsClient.cpp]: I think this at least the getSensors call can be supported, with an implementation like the one proposed. 
```suggestion
    if (!m_externalConnection) { 
        return metadataVector.size();
    } else {
        std::lock_guard<std::mutex>  guard(m_streamingPort.dataMutex);
        m_streamingPort.updateTimeoutStatus();
        if (m_streamingPort.status == yarp::dev::MAS_OK)  {
            return measurementsVector.measurements.size()
        } else {
            return 0;
        }
    }    
```
But we need to add `measurementsVector` as an argument.

**Pull request merged**